src/aws/flb_aws_util.c

```diff
@@ -322,8 +328,32 @@ struct flb_http_client *request_do(struct flb_aws_client *aws_client,
         flb_warn("[aws_http_client] failed to increase max response buffer size");
     } 
 
-    /* Add AWS Fluent Bit user agent */
+    /* Set AWS Fluent Bit user agent */
+    env = aws_client->upstream->config->env;
+    buf = (char *) flb_env_get(env, "FLB_AWS_USER_AGENT");
+    if (buf == NULL) {
+        if (getenv(AWS_ECS_METADATA_URI) != NULL) {
+            user_agent = "ecs";
+        } 
+        else {
+            buf = (char *) flb_env_get(env, "k8s");
```

Comment:
I checked and it seems this is set at startup in the k8s filter init: https://github.com/fluent/fluent-bit/blob/master/plugins/filter_kubernetes/kubernetes.c#L207
https://github.com/fluent/fluent-bit/blob/master/plugins/filter_kubernetes/kube_meta.c#L227

Reply:
yeah so we use this to check if we are using k8s filter